### PR TITLE
Rust pcap writer

### PR DESCRIPTION
**Pull request by stevenengler**
Replaces the pcap writer with a rust version.

Changes:
- `packet_getHeaderSize()` no longer includes the ethernet frame header
    - all code has been updated to accommodate this (ie: re-add the ethernet header), and their behaviour should not change
- fixes a bug that corrupts the pcap file when capturing large packets
- captures only IP packets rather than ethernet frames (`LINKTYPE_RAW` instead of `LINKTYPE_ETHERNET`)
    - shadow doesn't deal with ethernet frames and writes only dummy data in the ethernet header, so doesn't makes sense to log this
- supports variable-length captures
- supports UDP packets

Closes: #1949

### CODE REVIEW

**Comment by robgjansen** [comment]
While working through this I'm wondering: are the header size changes here are consistent with the changes in #1763?

**Comment by robgjansen** [comment]
Ok I think I see now that you are trying to be consistent and include the ethernet header sizes here, and removing them should be done in a separate PR (as was attempted in #1763) since it could have performance implications.

**Review by robgjansen — APPROVED** [review]
Looks like a big improvement :)

File: src/main/host/network_interface.c [hunk 1]
```diff
@@ -711,12 +689,19 @@ NetworkInterface* networkinterface_new(Host* host, Address* address, guint64 bwD
...
+
+        /* Append trailing slash if not present */
+        if (!g_str_has_suffix(filename->str, "/")) {
+            g_string_append(filename, "/");
+        }
+
+        g_string_append_printf(filename, "%s-%s.pcap", address_toHostName(interface->address),
+                               address_toHostIPString(interface->address));
+
+        interface->pcap = pcapwriter_new(filename->str, 65535);
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply src/main/host/network_interface.c]: Added.

**Pull request merged**